skimage/transform/_geometric.py

```diff
@@ -477,10 +556,9 @@ def _setup_constraint_matrix(self, src, dst):
             raise ValueError('src.shape[0] must be equal or larger than 8.')
 
         # Center and normalize image points for better numerical stability.
-        try:
-            src_matrix, src = _center_and_normalize_points(src)
-            dst_matrix, dst = _center_and_normalize_points(dst)
-        except ZeroDivisionError:
+        src_matrix, src = _center_and_normalize_points(src)
+        dst_matrix, dst = _center_and_normalize_points(dst)
+        if np.any(np.isnan(src_matrix + dst_matrix)):
```

Comment:
Thanks for the [commit message](https://github.com/scikit-image/scikit-image/commit/41669f8d5171d9656036d5c0ab9b88d370faa04c) which explained this change. :blush: